### ISSUE TITLE
[UX] Do not shift/sort the room list if the user has their cursor hovering over it

### ISSUE THREAD

**Issue by ShadowJonathan** [element-hq/element-web#16805]
#### Is your suggestion related to a problem? Please describe.
In addition to rooms being positionally-sticky in the sidebar (#16127), rooms currently "bounce around a lot" when sorted on activity.

There is nothing wrong with this (inherently), but users can miss-click rooms when they shift "underneath" them, sometimes between the fraction of a section where the user prepares to click, and has actually clicked.

Personally, this is especially-annoying as i leave some rooms unread so i can read them later, and matrix currently does not have a "mark unread" button on specific messages.

#### Describe the solution you'd like.
"Lock" the room list whenever the user has their mouse cursor over it, and then only unlock it under the following circumstances:
- The user's cursor has left the room list for more than a full second.
- The user's cursor hasn't moved for more than 10 seconds.

This effectively allows the user to, invisibly, "more easily" click rooms in an active room list.
